### PR TITLE
SCP Updates

### PR DESCRIPTION
**Pull request by tichys**
SCP - 035

SCP - 080 

SCP - 939

These various scp addons will expand gameplay

### CODE REVIEW

File: code/modules/SCP/SCPs/SCP-035.dm [hunk 1]
```diff
@@ -0,0 +1,200 @@
+GLOBAL_LIST_EMPTY(scp035s)
+/obj/item/clothing/mask/scp_035
+	name = "SCP-035"
+	icon = 'icons/SCP/scp-035.dmi'
+	icon_state = "scp035_0"
+	item_state = "scp035_0"
+	randpixel = FALSE
+	var/weeping = FALSE
+	var/tradegy = FALSE
+	var/mob/living/scp_035/brain_client
```
Comment (EgorDinamit) [inline code/modules/SCP/SCPs/SCP-035.dm:10]: Is there any reason for 035 "brain mob" to be living?

File: code/modules/SCP/SCPs/SCP-035.dm [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+										'sounds/hallucinations/035_whispers_1.ogg',
+										'sounds/hallucinations/035_whispers_2.ogg',
+										'sounds/hallucinations/035_whispers_3.ogg',
+										'sounds/hallucinations/035_whispers_4.ogg')
+
+/obj/item/clothing/mask/scp_035/Initialize()
+	. = ..()
+	brain_client = new()
+	brain_client.mask = src
+	contents += brain_client
```
Comment (EgorDinamit) [inline code/modules/SCP/SCPs/SCP-035.dm]:
```suggestion
	brain_client.forceMove(src)
```

Comment (EgorDinamit) [reply code/modules/SCP/SCPs/SCP-035.dm]: Actually you don't even need that. Just add src as arg above, I'll leave a comment.

Comment (EgorDinamit) [inline code/modules/SCP/SCPs/SCP-035.dm]:
```suggestion
	brain_client = new(src)
	brain_client.mask = src
```

File: code/modules/SCP/SCPs/SCP-035.dm [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+		"tragedy mask", //Name (Should not be the scp desg, more like what it can be described as to viewers)
+		SCP_KETER, //Obj Class
+		"035", //Numerical Designation
+		SCP_PLAYABLE|SCP_ROLEPLAY
+	)
+
+/obj/item/clothing/mask/scp_035/examine(mob/user)
+	. = ..()
+	var/message
+	if(icon_state)
```
Comment (EgorDinamit) [inline code/modules/SCP/SCPs/SCP-035.dm]:
```suggestion
	if(!tragedy)
```
I assume that was the intention here.

File: code/modules/SCP/SCPs/SCP-035.dm [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+	weeping = FALSE
+	update_icon()
+	addtimer(CALLBACK(src, .proc/weepstart), rand(9.5, 114) SECONDS) //only 2 min max downtime, it outputs alot
+	if(prob(5))
+		tradegy = !tradegy
+
+/obj/item/clothing/mask/scp_035/proc/output_substance()
+	if(!weeping)
+		return
+	if(prob(100)) //output under the mask 4/5 of the time
```
Comment (EgorDinamit) [inline code/modules/SCP/SCPs/SCP-035.dm]: Why prob(100)..?

Comment (tichys) [reply code/modules/SCP/SCPs/SCP-035.dm]: meant to be 80

File: code/modules/SCP/SCPs/SCP-035.dm [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+
+	// begin the actual takeover
+	to_chat(H, "<span class='scp035'>Your vision blurs as your mind is flooded with ancient memories that are not your own. They pour into your head, centuries of sinister history and malevolent thoughts.</span>")
+
+	addtimer(CALLBACK(src, .proc/continue_corruption, H), 30 SECONDS)
+
+/obj/item/clothing/mask/scp_035/proc/continue_corruption(mob/living/carbon/human/H)
+	to_chat(H, "<span class='scp035'>You feel your personality unraveling, your memories and identity drained away. It's like you are being hollowed out from the inside, everything that makes you who you are scraped away bit by bit.</span>")
+
+	canremove = FALSE // stop them from taking off the mask
```
Comment (EgorDinamit) [inline code/modules/SCP/SCPs/SCP-035.dm:124]: This is already done when putting mask on.

File: code/modules/SCP/SCPs/SCP-035.dm [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+
+	addtimer(CALLBACK(src, .proc/finish_corruption, H), 30 SECONDS)
+
+/obj/item/clothing/mask/scp_035/proc/finish_corruption(mob/living/carbon/human/H)
+
+	H.set_species("SCP-035-1") // changes appearance
+	H.real_name = "SCP-035"
+	H.name = H.real_name
```
Comment (EgorDinamit) [inline code/modules/SCP/SCPs/SCP-035.dm]: There's a proc for that. Also, wouldn't it be SCP-035-1? Besides, is there any point in changing names and species? Isn't it supposed to be *somewhat* stealthy?

File: code/modules/SCP/SCPs/SCP-035.dm (hunk 5 above)
Comment (tichys) [reply code/modules/SCP/SCPs/SCP-035.dm:124]: incorrect, when you put on the mask it starts the corruption and then after 30 seconds it calls this proc to continue the corruption

File: code/modules/SCP/SCPs/SCP-035.dm [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+	H.faction = "SCP"
+
+	// delete the original mind/brain and transfer SCP-035 mind into body
+	var/datum/mind/original_mind = H.mind
+	original_mind.current.ghostize()
+	qdel(original_mind)
```
Comment (EgorDinamit) [inline code/modules/SCP/SCPs/SCP-035.dm]: Just ghostize the mob, there's no need to delete their mind.

File: code/modules/SCP/SCPs/SCP-035.dm [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+
+	H.mind_initialize() // give it a new empty mind
+
+	var/mob/living/scp_035/scp = new(src)
```
Comment (EgorDinamit) [inline code/modules/SCP/SCPs/SCP-035.dm]: Why are you making ANOTHER? Isn't the mob created on init?

File: code/modules/SCP/SCPs/SCP-035.dm [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+
+
+//BRAIN AND ABILITIES
+
+/mob/living/scp_035
+	name = "SCP-035"
+	var/obj/item/clothing/mask/scp_035/mask
+
+//THE CORROSIVE LIQUID
+/datum/reagent/blood/scp_035_substance
```
Comment (EgorDinamit) [inline code/modules/SCP/SCPs/SCP-035.dm:163]: If you aren't using any part of the blood reagent, why make it blood subtype..?

File: code/modules/SCP/SCPs/SCP-080.dm [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+    status_flags = GODMODE // SCP-080 cannot be killed or gibbed
+    a_intent = I_HELP // SCP-080 does not have aggressive intent
+
+    // SCP-080 behavior flags
+    var/induce_fear = TRUE // Can SCP-080 induce fear?
+    var/fear_radius = 7 // Radius within which SCP-080 can affect characters
+    var/fear_strength = 5 // Strength of the fear effect
+    var/list/affected_mobs = list()
+    var/fear_message_cooldown = FEAR_MESSAGE_COOLDOWN
+/mob/living/scp080/Initialize()
```
Comment (EgorDinamit) [inline code/modules/SCP/SCPs/SCP-080.dm]:
```suggestion
    var/fear_message_cooldown = FEAR_MESSAGE_COOLDOWN

/mob/living/scp080/Initialize()
```
:)

File: code/modules/SCP/SCPs/SCP-080.dm [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+	. = ..()
+
+	SCP = new /datum/scp(
+		src, // Ref to actual SCP atom
+		"eery shadow", //Name (Should not be the scp desg, more like what it can be described as to viewers)
+		SCP_KETER, //Obj Class
+		"080", //Numerical Designation
+		SCP_PLAYABLE|SCP_ROLEPLAY
+	)
+/mob/living/scp080/Life(delta_time, times_fired)
```
Comment (EgorDinamit) [inline code/modules/SCP/SCPs/SCP-080.dm]:
```suggestion
	)

/mob/living/scp080/Life(delta_time, times_fired)
```

File: code/modules/SCP/SCPs/SCP-035.dm (hunk 6 above)
Comment (tichys) [reply code/modules/SCP/SCPs/SCP-035.dm]: how are you going to be stealthy with a mask on your face

**Review by EgorDinamit — CHANGES_REQUESTED** [review]
All procs must follow code standards and be in upper case with each word (i.e. WhisperText(), not whisper_text()).

File: code/modules/SCP/SCPs/SCP-035.dm [hunk 12]
```diff
@@ -0,0 +1,197 @@
+GLOBAL_LIST_EMPTY(scp035s)
+/obj/item/clothing/mask/scp_035
+	name = "SCP-035"
+	icon = 'icons/SCP/scp-035.dmi'
+	icon_state = "scp035_0"
+	item_state = "scp035_0"
+	randpixel = FALSE
+	var/weeping = FALSE
+	var/tradegy = FALSE
```
Comment (EgorDinamit) [inline code/modules/SCP/SCPs/SCP-035.dm:9]: Really now.
```suggestion
	var/tragedy = FALSE
```

File: code/modules/SCP/SCPs/SCP-035.dm (hunk 5 above)
Comment (EgorDinamit) [reply code/modules/SCP/SCPs/SCP-035.dm:124]: image

Comment (EgorDinamit) [reply code/modules/SCP/SCPs/SCP-035.dm:124]: canremove is set to FALSE when equipped. That's what I was talking about.

File: code/modules/SCP/SCPs/SCP-035.dm [hunk 13]
```diff
@@ -0,0 +1,197 @@
...
+/obj/item/clothing/mask/scp_035/proc/finish_corruption(mob/living/carbon/human/H)
+
+	H.set_species("SCP-035-1") // changes appearance
+	H.real_name = "SCP-035-1"
+	H.name = H.real_name
+	H.faction = "SCP"
+
+	// delete the original mind/brain and transfer SCP-035 mind into body
+	var/datum/mind/original_mind = H.mind
+	original_mind.current.ghostize()
```
Comment (EgorDinamit) [inline code/modules/SCP/SCPs/SCP-035.dm]: Might as well just do this.
```suggestion
	H.ghostize()
```

File: code/modules/SCP/SCPs/SCP-035.dm [hunk 14]
```diff
@@ -0,0 +1,197 @@
...
+
+	H.mind_initialize() // give it a new empty mind
+
+	H.mind.transfer_to(scp)
```
Comment (EgorDinamit) [inline code/modules/SCP/SCPs/SCP-035.dm]: Should be the other way around - you transfer 035's mind to the mob.

File: code/modules/SCP/SCPs/SCP-035.dm [hunk 15]
```diff
@@ -0,0 +1,197 @@
...
+
+	to_chat(scp, "<span class='scp035'>At last, a vessel to inhabit and control! I will make excellent use of this body... Finally, after years of waiting, I have a form to interact with the world once more. This mind is now mine, and I will not waste this opportunity.</span>")
```
Comment (EgorDinamit) [inline code/modules/SCP/SCPs/SCP-035.dm]: Mind is transferred, so H is now our 035 "brain mob".
```suggestion
	to_chat(H, "<span class='scp035'>At last, a vessel to inhabit and control! I will make excellent use of this body... Finally, after years of waiting, I have a form to interact with the world once more. This mind is now mine, and I will not waste this opportunity.</span>")
```

File: code/modules/SCP/SCPs/SCP-035.dm (hunk 14 above)
Comment (EgorDinamit) [inline code/modules/SCP/SCPs/SCP-035.dm]:
```suggestion
	brain_client.mind.transfer_to(H)
```

File: code/game/objects/effects/decals/Cleanable/humans.dm [hunk 16]
```diff
@@ -292,6 +292,9 @@ var/global/list/image/splatter_cache=list()
 /obj/effect/decal/cleanable/mucus/proc/set_dry()
 	dry = TRUE
 
+/obj/effect/decal/cleanable/mucus/update_icon()
+	. = ..()
```
Comment (Tsurupeta) [inline code/game/objects/effects/decals/Cleanable/humans.dm:296]: Ok but why

Comment (tichys) [reply code/game/objects/effects/decals/Cleanable/humans.dm:296]: cause you have to call your parents sometimes

Comment (Tsurupeta) [reply code/game/objects/effects/decals/Cleanable/humans.dm:296]: I just called them, they're doing fine. Why do you need to call parents implicitly if they're already being called without this piece of code???

**Pull request closed without merging**